src/context/AuthContext.js

```diff
@@ -0,0 +1,33 @@
+import axios from "axios";
+const { createContext, useState, useContext } = require("react");
+
+const AuthContext = createContext();
+
+const AuthProvider = ({ children }) => {
+    const [firstName, setFirstName] = useState("");
+    const [lastName, setLastName] = useState("");
+    const [encodedToken, setEncodedToken] = useState(null);
```

Comment:
Instead of separate states you can have these as useReducer

Reply:
ok I will do that